tt_eager/tt_dnn/op_library/composite/composite_ops.cpp

```diff
@@ -1308,6 +1309,136 @@ Tensor repeat(const Tensor& input_a, const Shape& shape_b, const MemoryConfig& o
+        // auto& input_shape = input_a.shape();
+        // int size = input_shape[0] * input_shape[1] * input_shape[2] * input_shape[3];
+        Tensor tindex = tt::numpy::index_hw<bfloat16>(input_shape, DataType::BFLOAT16);
+        Tensor max_val = reduce(input_a, ReduceOpMath::MAX, ReduceOpDim::HW);
+        Tensor max_tensor = zeros_like(input_a, output_mem_config);
+        max_tensor = bcast(max_tensor, max_val,  BcastOpMath::ADD, BcastOpDim::HW, output_mem_config);
+        Tensor cmp_results = eq(input_a, max_tensor, std::nullopt, output_mem_config);
+        Tensor max_indices =  mul(cmp_results ,tindex, std::nullopt, output_mem_config);
+        Tensor result = full_like(max_indices, size);
+        result = where(eqz(max_indices), result, max_indices);
+        result = reduce(result, ReduceOpMath::MIN, ReduceOpDim::HW);
+        Tensor res_index = zeros_like(result, output_mem_config);
+        result = where(eq(result, full_like(result, size)), res_index, result);
+        res_index = bcast(res_index, result,  BcastOpMath::ADD, BcastOpDim::HW, output_mem_config);
+        return res_index;
+    }
+
+}
+Tensor argmax(const Tensor& input_a, int dim, const MemoryConfig& output_mem_config /* = operation::DEFAULT_OUTPUT_MEMORY_CONFIG */) {
+    return operation::decorate_as_composite(__func__, _argmax)(input_a, dim, output_mem_config);
+}
+
+
+// Argmax returns the index of maximum element in the tensor
+Tensor _argmin(const Tensor& input_a, int dim, const MemoryConfig& output_mem_config) {
+
+    auto& input_shape = input_a.shape();
+    int size = input_shape[0] * input_shape[1] * input_shape[2] * input_shape[3];
+    TT_ASSERT((input_shape[0] == 1 && input_shape[1] == 1), "Unsupported shapes, supported shapes [1, 1, N, M]");
+
+    if (dim == 3 || dim == -1)
+    {
+        Tensor tindex = tt::numpy::index_width<bfloat16>(input_shape, DataType::BFLOAT16);
+        Tensor min_val = reduce(input_a, ReduceOpMath::MIN, ReduceOpDim::W);
+        Tensor min_tensor = zeros_like(input_a, output_mem_config);
+        min_tensor = bcast(min_tensor, min_val,  BcastOpMath::ADD, BcastOpDim::W, output_mem_config);
+        Tensor cmp_results = eq(input_a, min_tensor, std::nullopt, output_mem_config);
+        Tensor min_indices =  mul(cmp_results ,tindex, std::nullopt, output_mem_config);
+        Tensor midx = full_like(min_indices, size);
+        Tensor result = where(eqz(min_indices), midx, min_indices);
+        result = reduce(result, ReduceOpMath::MIN, ReduceOpDim::W);
+        Tensor res_index = zeros_like(result, output_mem_config);
+        result = where(eq(result, full_like(result, size)), res_index, result);
+        res_index = bcast(res_index, result,  BcastOpMath::ADD, BcastOpDim::W, output_mem_config);
+        std::vector<int64_t> permute_dims = {0, 1, 3, 2};
+        Tensor transpose_res = permute(res_index,permute_dims);
```

Comment:
You should scope/dellocate your tensors to reduce memory usage.
```
         Tensor min_val = reduce(input_a, ReduceOpMath::MIN, ReduceOpDim::W);
         Tensor min_tensor = zeros_like(input_a, output_mem_config);
         min_tensor = bcast(min_tensor, min_val,  BcastOpMath::ADD, BcastOpDim::W, output_mem_config);
         min_val.deallocate();
         Tensor cmp_results = eq(input_a, min_tensor, std::nullopt, output_mem_config);
         min_tensor.deallocate();
         Tensor tindex = tt::numpy::index_width<bfloat16>(input_shape, DataType::BFLOAT16);
         Tensor min_indices =  mul(cmp_results ,tindex, std::nullopt, output_mem_config);
         // tindex not on device
         // tindex.deallocate();
         Tensor midx = full_like(min_indices, size);
         Tensor result = where(eqz(min_indices), midx, min_indices);
         min_indices.deallocate();
         // midx not on device
         // midx.deallocate();
         result = reduce(result, ReduceOpMath::MIN, ReduceOpDim::W);
         Tensor res_index = zeros_like(result, output_mem_config);
         result = where(eq(result, full_like(result, size)), res_index, result);
         res_index = bcast(res_index, result,  BcastOpMath::ADD, BcastOpDim::W, output_mem_config);
         result.deallocate();
         std::vector<int64_t> permute_dims = {0, 1, 3, 2};
         Tensor transpose_res = permute(res_index,permute_dims);
```

Reply:
Updated the code